src/Graph.ts

```diff
@@ -325,12 +339,23 @@ export default class Graph {
 				timeStart('phase 2');
 
 				this.link();
-				const { orderedModules, dynamicImports } = this.analyseExecution(entryModules);
+
+				const {
+					orderedModules,
+					dynamicImports,
+					preservedModules
+				} = this.analyseExecution(entryModules, preserveModules);
+
 				dynamicImports.forEach(dynamicImportModule => {
 					if (entryModules.indexOf(dynamicImportModule) === -1)
 						entryModules.push(dynamicImportModule);
 				});
 
+				preservedModules.forEach(module => {
+					if (entryModules.indexOf(module) === -1)
+						entryModules.push(module);
+				});
```

Comment:
This is just "all modules" right? If so it should be possible to generate from Object.values(this.graph) + this.externals?

Reply:
removed with your other suggested change.